### PR TITLE
Implement HostQR code

### PR DESCRIPTION
**Pull request by crystalsong2021**
## Description

Adding fonts. For usage -> `import { fonts } from '../infrastructure/fonts.js'; fontFamily: fonts.bodyBold,`
Adding more code to QR code 
Screen Shot 2021-12-23 at 12 47 30 PM

## How to test

use the QR scanner to test out the HostQR and able to pass sessionId and HostName
https://snack.expo.dev/@sugarexpo/380485

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -56,13 +89,16 @@ export default function App() {
...
-      </ThemeProvider>
+      { !fontsLoaded
+        ? <AppLoading />
+        : (<ThemeProvider theme={theme}>
+          <NavigationContainer>
+            <HomeScreen />
+          </NavigationContainer>
+          {/* <DummyComponent /> */}
+        </ThemeProvider>)
+      }
```
Comment (ashequl93) [inline App.js:100]: Hi, I did not understand the conditional here. If font is not loaded, then you're returning `<AppLoading/>` else you are returning the actual app code, is it so? Why are you returning the `<ThemeProvider />` wrapped component if the font is not loaded?

File: package 2.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  "devDependencies": {
+    "@babel/core": "^7.12.9",
+    "eslint": "^7.32.0",
+    "eslint-config-hackreactor": "github:reactorcore/eslint-config-hackreactor",
+    "eslint-plugin-react": "^7.27.1",
+    "eslint-plugin-react-native": "^4.0.0",
+    "prettier": "^2.5.1"
+  },
+  "private": true
+}
```
Comment (ashequl93) [inline package 2.json]: Why do we need a package2.json file?

File: package 3.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  "devDependencies": {
+    "@babel/core": "^7.12.9",
+    "eslint": "^7.32.0",
+    "eslint-config-hackreactor": "github:reactorcore/eslint-config-hackreactor",
+    "eslint-plugin-react": "^7.27.1",
+    "eslint-plugin-react-native": "^4.0.0",
+    "prettier": "^2.5.1"
+  },
+  "private": true
+}
```
Comment (ashequl93) [inline package 3.json]: Why do we need a package3.json file?

File: src/infrastructure/fonts.js [hunk 4]
```diff
@@ -1,3 +1,27 @@
...
+export const fontSizes = {
+  caption: '12px',
+  button: '14px',
+  body: '16px',
+  title: '20px',
+  h5: '24px',
+  h4: '34px',
+  h3: '45px',
+  h2: '56px',
+  h1: '112px',
```
Comment (ashequl93) [inline src/infrastructure/fonts.js]: Do we need these tags here? If not, can you remove them please?

**Review by ashequl93 — CHANGES_REQUESTED** [review]
Kindly look at the comments i made.

File: App.js (hunk 1 above)
Comment (tm2838) [reply App.js:100]: I think the wrapped component is rendered if font is loaded?

**Review by tm2838 — COMMENTED** [review]
The QR code scanner looks really cool! Glad we have this feature. Great work!
I'm curious about the `package 2/3.json` too. And regarding file structure, I believe `HostQR` is for 'current session - QR' tab and `GuestQR` is for `join` tab, could we move the two files to the respective subfolders under `features`? and probably `GuestQR` can be directly imported into `Join.Scree.js` ?

File: App.js (hunk 1 above)
Comment (ashequl93) [reply App.js:100]: Got it. Thanks.

Comment (crystalsong2021) [reply App.js:100]: yes. If load the fonts first, then load the <ThemeProvider />

File: package 2.json (hunk 2 above)
Comment (crystalsong2021) [reply package 2.json]: I was curious myself too. I deleted it and updated it.

File: package 3.json (hunk 3 above)
Comment (crystalsong2021) [reply package 3.json]: same. delete and update now.

File: src/infrastructure/fonts.js (hunk 4 above)
Comment (crystalsong2021) [reply src/infrastructure/fonts.js]: no..done.

**Review by ashequl93 — APPROVED** [review]
Thanks for the quick fix.

**Review by tm2838 — APPROVED** [review]

**Pull request merged**